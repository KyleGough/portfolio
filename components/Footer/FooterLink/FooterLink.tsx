import Link from 'next/link';
import React from 'react';

interface FooterLinkProps {
  children: React.ReactNode;
  href: string;
}

export const FooterLink: React.FC<FooterLinkProps> = ({ children, href }) => {
  return (
    <li>
      <Link href={href} className="p-2 hover:text-nav-hover">
        {children}
      </Link>
    </li>
  );
};
