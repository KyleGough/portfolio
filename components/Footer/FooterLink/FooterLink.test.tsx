import { render, screen } from '@testing-library/react';
import React from 'react';

import { FooterLink } from './FooterLink';

describe('FooterLink component', () => {
  it('renders', () => {
    const { container } = render(<FooterLink href="/">Homepage</FooterLink>);
    expect(screen.getByText('Homepage')).toBeVisible();
    expect(screen.getByRole('link')).toHaveAttribute('href', '/');
    expect(container).toMatchSnapshot();
  });
});
