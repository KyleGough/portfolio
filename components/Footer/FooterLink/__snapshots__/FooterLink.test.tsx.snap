// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLink component renders 1`] = `
<div>
  <li>
    <a
      class="p-2 hover:text-nav-hover"
      href="/"
    >
      Homepage
    </a>
  </li>
</div>
`;
